src/Jobs/Recruit.Vacancies.Jobs/DomainEvents/Handlers/Provider/ProviderBlockedDomainEventHandler.cs

```diff
@@ -0,0 +1,129 @@
+using Esfa.Recruit.Vacancies.Client.Domain.Repositories;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.QueryStore;
+using Esfa.Recruit.Vacancies.Client.Infrastructure.QueryStore.Projections.EditVacancyInfo;
+using Microsoft.Extensions.Logging;
+
+namespace Esfa.Recruit.Vacancies.Jobs.DomainEvents.Handlers.Provider
+{
+    public class ProviderBlockedDomainEventHandler : DomainEventHandler, IDomainEventHandler<ProviderBlockedEvent>
+    {
+        private readonly IQueryStoreReader _queryStoreReader;
+        private readonly IVacancyQuery _vacancyQuery;
+        private readonly IMessaging _messaging;
+
+        public ProviderBlockedDomainEventHandler(
+            IQueryStoreReader queryStoreReader,
+            IVacancyQuery vacancyQuery,
+            IMessaging messaging,
+            ILogger<ProviderBlockedDomainEventHandler> logger) : base(logger)
+        {
+            _queryStoreReader = queryStoreReader;
+            _vacancyQuery = vacancyQuery;
+            _messaging = messaging;
+        }
+
+        public async Task HandleAsync(string eventPayload)
+        {
+            var eventData = DeserializeEvent<ProviderBlockedEvent>(eventPayload);
+
+            var tasks = new List<Task>();
+
+            var providerInfo = await _queryStoreReader.GetProviderVacancyDataAsync(eventData.Ukprn);
+            tasks.AddRange(RaiseEventsToRevokePermission(providerInfo.Employers, eventData.Ukprn));
+
+            var vacancies = await _vacancyQuery.GetVacanciesAssociatedToProvider(eventData.Ukprn);
+
+            tasks.AddRange(RaiseEventsToUpdateVacancies(vacancies, eventData.QaVacancyUser));
+
+            tasks.Add(RequestProviderCommunication(eventData.ProviderName));
+
+            tasks.AddRange(RequestEmployerCommunications(vacancies));
+
+            //TODO update employer and provider dashboard
+
+            //TODO update providereditinfo... delete Employers list ???
+
+            await Task.WhenAll(tasks);
+        }
+
+        private List<Task> RaiseEventsToRevokePermission(IEnumerable<EmployerInfo> employers, long ukprn)
+        {
+            var tasks = new List<Task>();
+            foreach (var employer in employers)
+            {
+                foreach (var legalEntity in employer.LegalEntities)
+                {
+                    var providerBlockedOnLegalEntityEvent = new ProviderBlockedOnLegalEntityEvent()
+                    {
+                        Ukprn = ukprn,
+                        EmployerAccountId = employer.EmployerAccountId,
+                        LegalEntityId = legalEntity.LegalEntityId
+                    };
+
+                    tasks.Add(_messaging.PublishEvent(providerBlockedOnLegalEntityEvent));
+                }
+            }
+            return tasks;
+        }
+
+        private List<Task> RaiseEventsToUpdateVacancies(IEnumerable<ProviderVacancySummary> vacancies, VacancyUser qaVacancyUser)
+        {
+            var tasks = new List<Task>();
```

Comment:
I cannot use TransferVacanciesFromProvider because it only looks at provider owned vacancies and I need to deal with employer owned vacancy as well.
There is no way to tell when RaiseEventsToRevokePermission is completed as it raises multiple events which are handled asynchronously. 
For this requirement it was agreed that each task will be done asynchronously so it can be retried individually or be dealt with on each failure basis.